File: apps/api/src/Stoxly.Api/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Stoxly.Api.Models;

public enum TransactionType
{
    BUY,
    SELL,
}

[Table("transactions")]
public class Transaction
{
    [Key]
    [Column("id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    [Required]
    [Column("portfolio_id")]
    public Guid PortfolioId { get; set; }

    [Required]
    [Column("symbol")]
    [MaxLength(20)]
    public string Symbol { get; set; } = string.Empty;

    [Required]
    [Column("type")]
    [MaxLength(4)]
    public TransactionType Type { get; set; }

    [Required]
    [Column("quantity")]
    public decimal Quantity { get; set; }

    [Required]
    [Column("price")]
    public decimal Price { get; set; }

    [Column("fee")]
    public decimal Fee { get; set; }

    [Required]
    [Column("trade_date")]
    public DateTime TradeDate { get; set; }

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation property
    public Portfolio? Portfolio { get; set; }
}

File: apps/web/src/types/transaction.ts
export type TransactionType = "BUY" | "SELL";

export interface TransactionResponse {
  id: string;
  portfolioId: string;
  portfolioName: string | null;
  symbol: string;
  type: TransactionType;
  quantity: number;
  price: number;
  fee: number;
  total: number;
  tradeDate: string;
  notes: string | null;
  createdAt: string;
}

export interface CreateTransactionRequest {
  symbol: string;
  type: TransactionType;
  quantity: number;
  price: number;
  fee?: number;
  tradeDate: string;
  notes?: string;
}

export interface UpdateTransactionRequest {
  portfolioId: string;
  fee?: number;
  notes?: string | null;
}

// Simulation trading

export interface SimulationBuyRequest {
  portfolioId: string;
  symbol: string;
  quantity: number;
  notes?: string;
}

export interface SimulationSellRequest {
  portfolioId: string;
  symbol: string;
  quantity: number;
  notes?: string;
}

export interface SimulationTradeResponse {
  transactionId: string;
  symbol: string;
  side: "BUY" | "SELL";
  quantity: number;
  price: number;
  total: number;
  fee: number;
  executedAt: string;
  remainingCashBalance: number;
  notes: string | null;
}
